src/textual/app.py

```diff
@@ -1782,9 +1791,17 @@ def render(renderable: RenderableType) -> list[Segment]:
     def _handle_exception(self, error: Exception) -> None:
         """Called with an unhandled exception.
 
+        Always results in the app exiting.
+
         Args:
             error: An exception instance.
         """
+        # If we're running via pilot and this is the first exception encountered,
+        # take note of it so that we can re-raise for test frameworks later.
+        if self.is_headless and self._exception is None:
```

Comment:
Why the `is_headless` guard here? There are a few reasons why `is_headless` may be True, unrelated to testing.

Reply:
Because I didn't know that 😄 I was looking for a "if running via pilot" flag I could use for this guard.

What are the other reasons?

Reply:
Docs screenshots, automation, etc.

Reply:
Ok, well we probably want exceptions to be raised up to any code which does automated things such as the docs screenshots. If an exception occurs during a screenshot rendering, we probably want to know about it rather than continuing silently with a broken screenshot.

Reply:
Agree. But we want to be sure not to raise exceptions if we have generated a nice error message.

Reply:
I think we still want to raise exceptions to prevent silent issues